sim: Fix MemoryError at system call

RashidAGP opened the pull request:
This fixes an issue with inconsistent data type conversion in the `read` system call.  
Fixes #1996

Commit:
Fixed. MemoryError at systemcall

erin-le commented:
@RashidAGP: Thank you for your contribution! Could you update your PR and commit titles to have the format specified on pages 32 and 33 of the following guide? https://bootcamp.gem5.org/#06-Contributing/01-contributing. It is also recommended to install pre-commit, which is discussed on pages 29-31.

Essentially, you need to have one or more tags, a colon, and then the title of your PR. A list of tags can be found here: https://github.com/gem5/gem5/blob/stable/MAINTAINERS.yaml. For this PR, the PR and commit titles should be something like `sim: MemoryError at systemcall`.

Also, could you change your target branch to `develop` instead of `stable`?

erin-le commented:
@RashidAGP: Hello, just a reminder about this PR. If you have time, could you change the target branch of this PR to `develop` and change the PR and commit titles to something along the lines of `sim: MemoryError at systemcall`? You should be able to edit the PR title and target branch by clicking `Edit` to the right of the PR title.
![Screenshot from 2025-03-05 12-05-03](https://github.com/user-attachments/assets/e693d4d1-450e-456d-882b-1d6a05274807)

erin-le commented:
@RashidAGP Just a heads up, I am going to push the requested changes to this PR so we can get this into develop. Thanks again for your contribution!

erin-le committed:
sim: use uint64_t for size in syscall_emul_buf.hh

This commit changes the type of the size parameter in
syscall_emul_buf.hh from size_t in the previous commit
to uint64_t.

giactra commented:
One of the commits in the PR (the original) does not have a valid commit title as it does not contain a tag.
I understand @erin-le you didn't want to modify the original author commit. It makes sense.

@powerjg what is your take here? Shall we merge this as is or shall @erin-le create a new PR?

powerjg commented:
We can squash and merge and fix the commit message there. There's no reason for this to be two commits anyway.
